src/plugins.ts

```diff
@@ -69,15 +69,53 @@ export function firebaseJWTPlugin(schema: Schema) {
  * @param schema Mongoose Schema
  */
 export function findOneOrThrow<T>(schema: Schema) {
-  schema.statics.findOneOrThrow = async function (query: FilterQuery<T>): Promise<T | null> {
+  schema.statics.findOneOrThrow = async function (
+    query: FilterQuery<T>,
+    errorArgs?: Partial<APIErrorConstructor>
+  ): Promise<T | null> {
     const results = await this.find(query);
     if (results.length === 0) {
       return null;
     } else if (results.length > 1) {
       throw new APIError({
         status: 500,
-        title: "findOne query returned multiple documents",
+        title: `${this.modelName}.findOne query returned multiple documents`,
         detail: `query: ${JSON.stringify(query)}`,
+        ...errorArgs,
+      });
+    } else {
+      return results[0];
+    }
+  };
+}
+
+/**
+ * This adds a static method `Model.findExactlyOne` to the schema. This or findOneOrThrow should replace `Model.findOne`
+ * in most instances.
+ * `Model.findOne` should only be used with a unique index, but that's not apparent from the docs. Otherwise you can wind
+ * up with a random document that matches the query. The returns the one matching document, or throws an exception if
+ * multiple or none are found.
+ * @param schema Mongoose Schema
+ */
+export function findExactlyOne<T>(schema: Schema) {
+  schema.statics.findExactlyOne = async function (
+    query: FilterQuery<T>,
+    errorArgs?: Partial<APIErrorConstructor>
+  ): Promise<T | null> {
+    const results = await this.find(query);
+    if (results.length === 0) {
+      throw new APIError({
+        status: 500,
```

Comment:
I first think `400 Not Found` when it has nothing to return, not 500. Just a thought, it is 'exactly one' so can be persuaded to keep it as is.

Reply:
Ooo that's fair!